pkg/api/v2/reporter/reporter.go

```diff
@@ -284,6 +293,8 @@ func (r *reporter) attemptUpdateStatus(ctx context.Context, client ReporterResou
 		if inputResourceFromRead, ok := resourceFromRead.(resources.InputResource); ok {
 			resourceToWrite = inputResourceFromRead
 			r.statusClient.SetStatus(resourceToWrite, statusToWrite)
```

Comment:
we probably want a new api beyond `r.statusClient.SetStatus` that also allows us to set the new fields we're adding e.g. `SetMessage` and `SetUpdatedAt`; or preferably a well named function that has us set all the fields at once

Reply:
see https://github.com/solo-io/solo-kit/pull/472/files#r882880743 for how we might want to implement these new functions (on a status client, not a new client)

Reply:
Added!